<h1 style="text-align: center;">[Dekoruma Analysis]</h1>
<h3 style="text-align: center;">[Elmar Leonard and Firsa Adam]</h3>

---

## **Section 1. Business Context**

**1.1 Context**

**1.2 Problem Statements**

**1.3 Key Objective**

## **Section 2. Data Understanding**

**2.1 General Information**

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import scipy

**Calling data**
> inserting all the data from csv into the notebook in form of dataframe

In [14]:
df_raw_orders = pd.read_csv(r"..\data\raw\dekoruma_orders.csv")
df_raw_products = pd.read_csv(r"..\data\raw\dekoruma_products.csv")
df_raw_stores = pd.read_csv(r"..\data\raw\dekoruma_stores.csv")

**2.2 Feature Information**

**Checking orders columns information**
> checking the data in the dataframe. Checking like this would allow us to see the connection between each tables of data and what is the column with the type of data inside of the column. making us know what should we casting later on if it doesnt fit with the correct information we need just yet.

In [15]:
df_raw_orders.info()

<class 'pandas.DataFrame'>
RangeIndex: 300000 entries, 0 to 299999
Data columns (total 9 columns):
 #   Column               Non-Null Count   Dtype
---  ------               --------------   -----
 0   order_id             300000 non-null  str  
 1   store_id             300000 non-null  str  
 2   sku_id               300000 non-null  str  
 3   order_date           300000 non-null  str  
 4   delivery_date        300000 non-null  str  
 5   installer_id         190275 non-null  str  
 6   installation_date    190275 non-null  str  
 7   installation_status  300000 non-null  str  
 8   assembly_fee         300000 non-null  int64
dtypes: int64(1), str(8)
memory usage: 51.2 MB


**Checking products columns information**
> checking the data in the dataframe. Checking like this would allow us to see the connection between each tables of data and what is the column with the type of data inside of the column. making us know what should we casting later on if it doesnt fit with the correct information we need just yet.

In [16]:
df_raw_products.info()

<class 'pandas.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   sku_id                1500 non-null   str  
 1   product_name          1500 non-null   str  
 2   category              1500 non-null   str  
 3   is_assembly_required  1500 non-null   bool 
dtypes: bool(1), str(3)
memory usage: 106.5 KB


**Checking stores columns information**
> checking the data in the dataframe. Checking like this would allow us to see the connection between each tables of data and what is the column with the type of data inside of the column. making us know what should we casting later on if it doesnt fit with the correct information we need just yet.

In [17]:
df_raw_stores.info()

<class 'pandas.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   store_id      40 non-null     str  
 1   store_name    40 non-null     str  
 2   channel_type  40 non-null     str  
dtypes: str(3)
memory usage: 2.5 KB


**2.3 Statistics Summary**

## **Section 3. Data Cleaning**

**3.1 Missing Values**

**Checking missing value data using isnull method**
> in this step we would be able to see how many null data inside each table so we can conclude why the data is missing and what should we do with it

In [29]:
print("=== ORDERS TABLE ===")
print(df_raw_orders.isnull().sum())

print("\n=== PRODUCTS TABLE ===")
print(df_raw_products.isnull().sum())

print("\n=== STORES TABLE ===")
print(df_raw_stores.isnull().sum())

=== ORDERS TABLE ===
order_id                    0
store_id                    0
sku_id                      0
order_date                  0
delivery_date               0
installer_id           109725
installation_date      109725
installation_status         0
assembly_fee                0
dtype: int64

=== PRODUCTS TABLE ===
sku_id                  0
product_name            0
category                0
is_assembly_required    0
dtype: int64

=== STORES TABLE ===
store_id        0
store_name      0
channel_type    0
dtype: int64


**In-Depth Dive into Missing Values**
> in this step we would dive deeper on the missing value on orders table to decide what type of missing value it is. is it MNAR, MAR, or MCAR.

**As we can see both missing value is on installer_id and installer date so we could check if the data missing is on the same row as it have the same amount of missing values**

In [33]:
df_raw_orders[(df_raw_orders["installer_id"].isnull())&(df_raw_orders["installation_date"].isnull())]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee
4,ORD-DKR00000005,DKR-019,SKU-FURN-01500,2023-12-15 12:00:00,2023-12-22 12:00:00,NaN,NaN,No Assembly Service,0
6,ORD-DKR00000007,DKR-025,SKU-FURN-01195,2023-02-14 09:00:00,2023-02-19 09:00:00,NaN,NaN,No Assembly Required,0
8,ORD-DKR00000009,DKR-022,SKU-FURN-00766,2023-07-29 13:00:00,2023-08-02 13:00:00,NaN,NaN,No Assembly Required,0
9,ORD-DKR00000010,DKR-026,SKU-FURN-01470,2023-12-29 10:00:00,2024-01-02 10:00:00,NaN,NaN,No Assembly Required,0
13,ORD-DKR00000014,DKR-020,SKU-FURN-01242,2023-07-02 14:00:00,2023-07-05 14:00:00,NaN,NaN,No Assembly Required,0
...,...,...,...,...,...,...,...,...,...
299991,ORD-DKR00299992,DKR-036,SKU-FURN-00004,2023-04-14 10:00:00,2023-04-18 10:00:00,NaN,NaN,No Assembly Service,0
299992,ORD-DKR00299993,DKR-019,SKU-FURN-01008,2023-10-07 16:00:00,2023-10-12 16:00:00,NaN,NaN,No Assembly Required,0
299996,ORD-DKR00299997,DKR-012,SKU-FURN-00579,2023-11-28 12:00:00,2023-12-04 12:00:00,NaN,NaN,No Assembly Required,0
299998,ORD-DKR00299999,DKR-031,SKU-FURN-01007,2023-09-06 11:00:00,2023-09-11 11:00:00,NaN,NaN,No Assembly Required,0


**from the result we can see that the row that have missing values is still 109725 even after we use a condition to call rows that have missing values on both installer_id and installation_date. Other than that as we now knew both of them correlated we can see as well all of the data that have missing row have assembly_fee of 0, with this we can make a first assumption that the data is missing not at random but because it has no assembly fee.**

**Testing first assumption**

In [35]:
df_raw_orders[(df_raw_orders["installer_id"].isnull())&(df_raw_orders["installation_date"].isnull())&(df_raw_orders["assembly_fee"]>0)]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee


**from this code we can conclude that the first assumption is correct with no rows with null values if the assembly fee is larger than zero. now we can do a more in depth research using installation status of each type**

In [37]:
df_raw_orders["installation_status"].unique()

<ArrowStringArray>
[               'Success',    'No Assembly Service',   'No Assembly Required',
 'Failed - Missing Parts', 'Failed - Damaged Panel']
Length: 5, dtype: str

**Checking if there is a null value in Success Status**

In [38]:
df_raw_orders[(df_raw_orders["installation_status"]=="Success")&(df_raw_orders["installer_id"].isnull())]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee


**Checking if there is a null value in Failed - Missing Parts Status**

In [39]:
df_raw_orders[(df_raw_orders["installation_status"]=="Failed - Missing Parts")&(df_raw_orders["installer_id"].isnull())]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee


**Checking if there is a null value in Failed - Damage Panel Status**

In [40]:
df_raw_orders[(df_raw_orders["installation_status"]=="Failed - Damage Panel")&(df_raw_orders["installer_id"].isnull())]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee


**Checking if there is a null value in No Assembly Service Status**

In [41]:
df_raw_orders[(df_raw_orders["installation_status"]=="No Assembly Service")&(df_raw_orders["installer_id"].isnull())]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee
4,ORD-DKR00000005,DKR-019,SKU-FURN-01500,2023-12-15 12:00:00,2023-12-22 12:00:00,NaN,NaN,No Assembly Service,0
27,ORD-DKR00000028,DKR-021,SKU-FURN-01464,2023-10-18 12:00:00,2023-10-20 12:00:00,NaN,NaN,No Assembly Service,0
38,ORD-DKR00000039,DKR-021,SKU-FURN-00911,2023-05-15 14:00:00,2023-05-19 14:00:00,NaN,NaN,No Assembly Service,0
73,ORD-DKR00000074,DKR-035,SKU-FURN-00348,2023-02-03 13:00:00,2023-02-05 13:00:00,NaN,NaN,No Assembly Service,0
79,ORD-DKR00000080,DKR-032,SKU-FURN-01049,2023-03-21 10:00:00,2023-03-27 10:00:00,NaN,NaN,No Assembly Service,0
...,...,...,...,...,...,...,...,...,...
299967,ORD-DKR00299968,DKR-040,SKU-FURN-00748,2023-05-06 16:00:00,2023-05-10 16:00:00,NaN,NaN,No Assembly Service,0
299979,ORD-DKR00299980,DKR-039,SKU-FURN-00540,2023-12-11 19:00:00,2023-12-13 19:00:00,NaN,NaN,No Assembly Service,0
299981,ORD-DKR00299982,DKR-026,SKU-FURN-00812,2023-01-14 12:00:00,2023-01-16 12:00:00,NaN,NaN,No Assembly Service,0
299991,ORD-DKR00299992,DKR-036,SKU-FURN-00004,2023-04-14 10:00:00,2023-04-18 10:00:00,NaN,NaN,No Assembly Service,0


**Checking if there is a null value in No Assembly Required Status**

In [42]:
df_raw_orders[(df_raw_orders["installation_status"]=="No Assembly Required")&(df_raw_orders["installer_id"].isnull())]

,order_id,store_id,sku_id,order_date,delivery_date,installer_id,installation_date,installation_status,assembly_fee
6,ORD-DKR00000007,DKR-025,SKU-FURN-01195,2023-02-14 09:00:00,2023-02-19 09:00:00,NaN,NaN,No Assembly Required,0
8,ORD-DKR00000009,DKR-022,SKU-FURN-00766,2023-07-29 13:00:00,2023-08-02 13:00:00,NaN,NaN,No Assembly Required,0
9,ORD-DKR00000010,DKR-026,SKU-FURN-01470,2023-12-29 10:00:00,2024-01-02 10:00:00,NaN,NaN,No Assembly Required,0
13,ORD-DKR00000014,DKR-020,SKU-FURN-01242,2023-07-02 14:00:00,2023-07-05 14:00:00,NaN,NaN,No Assembly Required,0
15,ORD-DKR00000016,DKR-016,SKU-FURN-01458,2023-05-20 13:00:00,2023-05-22 13:00:00,NaN,NaN,No Assembly Required,0
...,...,...,...,...,...,...,...,...,...
299988,ORD-DKR00299989,DKR-031,SKU-FURN-01493,2023-02-07 10:00:00,2023-02-13 10:00:00,NaN,NaN,No Assembly Required,0
299989,ORD-DKR00299990,DKR-004,SKU-FURN-00577,2023-01-21 15:00:00,2023-01-23 15:00:00,NaN,NaN,No Assembly Required,0
299992,ORD-DKR00299993,DKR-019,SKU-FURN-01008,2023-10-07 16:00:00,2023-10-12 16:00:00,NaN,NaN,No Assembly Required,0
299996,ORD-DKR00299997,DKR-012,SKU-FURN-00579,2023-11-28 12:00:00,2023-12-04 12:00:00,NaN,NaN,No Assembly Required,0


**After looking more in-depth on the information we have we now can confidently conclude that the missing values is not random but because of a reason that is if the installation is not needed for that product or there is no assembly service for that product then the installer_id and installation_date would be filled with NaN**

**3.2 Duplicated Values**

**3.3 Identify Spelling Errors**

**3.4 Identify Anomaly Values**

## **Section 4. Analytics**

### **4.1 Question 1**

### **4.2 Question 2**

## **Section 5. Conclusion and Recommendation**

**5.1 Conclusion**

**5.2 Recommendation**